Example2/CPP_Abstraction/xt_table_info.cpp

```diff
@@ -739,12 +841,54 @@ void xt_free_table_info(struct xt_table_info *info)
 }
 
 
+/**
+* @brief Fuction for testing the abstraction -Modified function to use the C++ abstraction
+* 
+* Input Parameter:(e, info, base, newinfo)
+*    @param e - LinkedList used to store the row entries of the routing table 
+*    @param info - The network routing table.
+*    @param base - Pointer to the starting position of the routing table entries and was 
+*                  used with e (struct ipt_entry) to get the next table row. 
+*    @param newinfo - The new network routing table that gets returned as output
+* 
+* Output Parameter:(newinfo)  
+*     @param newinfo - the new table (abstraction.size, abstraction.hook_entry & abstraction.underflow are modified)      
+*         
+* @return (int) -- returning different error codes, possible output {0 -- success, a negative errno code, positive exit code on failure}.
+*/  
+static int compat_calc_entry(const struct ipt_entry *e,
```

Comment:
See the other comments -- this function should not be broken up. Instead, it should be used as an API of the new class.